librubyfmt/src/render_targets.rs

```diff
@@ -3,11 +3,58 @@ use crate::line_tokens::LineToken;
 use crate::types::{ColNumber, LineNumber};
 use std::collections::HashSet;
 
+fn insert_at<T>(idx: usize, target: &mut Vec<T>, input: &mut Vec<T>) {
+    let drain = input.drain(..);
+    let mut idx = idx;
+    for item in drain {
+        target.insert(idx, item);
+        idx += 1;
+    }
```

Comment:
```suggestion
    let mut tail = target.split_off(idx);
    input.append(input);
    input.append(&mut tail);
```

Reply:
The code as written will perform as many as log(N) allocations and 2N memcpys. The new code will perform at most 2 and at minimum 1 allocations, and 2 memcpys